libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/data/MapboxNavigationViewportDataSource.kt

```diff
@@ -434,16 +433,27 @@ class MapboxNavigationViewportDataSource(
      * @see [evaluate]
      */
     fun onRouteProgressChanged(routeProgress: RouteProgress) {
-        // todo abort if there's no route cached or differs from the one in rp
-        if (this.route == null) {
-            return
-        }
-
-        if (this.route?.isSameUuid(routeProgress.route) != true) {
+        val currentRoute = this.route
+        if (currentRoute == null) {
             LoggerProvider.logger.w(
                 Tag("MbxViewportDataSource"),
-                Message("Provided route and navigated route do not have the same UUID.")
+                Message(
+                    "You're calling #onRouteProgressChanged but you didn't call #onRouteChanged."
+                )
             )
+            clearProgressData()
+            return
+        } else if (!currentRoute.isSameRoute(routeProgress.route)) {
+            LoggerProvider.logger.e(
+                Tag("MbxViewportDataSource"),
```

Comment:
Could we extract this magic number into a `const`ant?

```suggestion
                Tag(TAG),
```

Reply:
I'll extract it :+1: